### PR TITLE
datadog-apm: deprecate datadog exporter

### PR DESCRIPTION
**Pull request by mabdinur**
# Description

Adds deprecation note to the Datadog Span Exporter.

## Type of change

Updates Readme

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

N/A

- [ ] Test A

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: exporter/opentelemetry-exporter-datadog/README.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-OpenTelemetry Datadog Exporter
+OpenTelemetry Datadog Span Exporter
 ==============================
```
Comment (codeboten) [inline exporter/opentelemetry-exporter-datadog/README.rst:2]:
```suggestion
OpenTelemetry Datadog Span Exporter
===================================
```

**Review by codeboten — COMMENTED** [review]
Pinging @majorgreys as the original author

**Comment by brettlangdon** [comment]
Once this PR is approved/merged is it possible to update the PyPI description for this package?

**Comment by codeboten** [comment]
> Once this PR is approved/merged is it possible to update the PyPI description for this package?

The pypi documentation will be updated when the next release of the package goes out.

**Comment by brettlangdon** [comment]
> The pypi documentation will be updated when the next release of the package goes out.

Yeah, my thoughts are if we don't do another release, could still be nice to update the notes.

Or maybe the right way is to do a new release even without code changes because we are marking the package as deprecated ?

**Review by brettlangdon — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**